pilot/pkg/bootstrap/server.go

```diff
@@ -168,13 +177,23 @@ type PilotArgs struct {
 // Server contains the runtime configuration for the Pilot discovery service.
 type Server struct {
 	mesh              *meshconfig.MeshConfig
-	serviceController *aggregate.Controller
+	ServiceController *aggregate.Controller
```

Comment:
Justify for making these public. 

doc comments are needed for every public variables.

Reply:
Are used from other packages - for some reasons all exposed fields in this class don't have 
doc comments. Since it's not an API we would expose outside pilot, and most of this is 
scheduled for deletion/deprecation - I don't think it's a huge priority.

Reply:
There are no exposed fields on the base version. I'm surprised we don't have a linter forcing some comments on public fields.
But if this is really soon to be deleted code... ok (though why do we need to add and expose new soon to be deleted code?)

Reply:
Having a struct exposing both private and public fields is an antipattern. Please add TODO to remove this code later.

Reply:
bootstrap/server.go is full of public fields - MeshArgs, ConsuleArgs, etc. 
We are planning to replace the v1 registries with new data structures and event-based code, and almost all the http-based envoy discovery will also go away once v2 is done ( target is 0.8)